### ISSUE TITLE
TYPE_CHECKING imports raise F401

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#1062]
In GitLab by @henryJack on Sep 7, 2017, 02:59

```
$ pip install  flake8

```

```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from my_module.test_class import MyClass

object: 'MyClass' = None
```
Flake 8 gives the following error in this scenario:

`my_module\test_class.py:4:5: F401 'my_module.test_class.MyClass' imported but unused`

My current work around is this:

```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from my_module.test_class import MyClass  # noqa

object: 'MyClass' = None
```

Any better solutions? Couldwe somehow automatically ignore all F401 errors within the `if TYPE_CHECKING:` block?

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Sep 7, 2017, 05:41

This should be reported to Pyflakes. I suspect the answer will be no.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Sep 7, 2017, 05:41

closed

**Comment by asottile** [comment]
In GitLab by @taion on Nov 22, 2017, 14:29

I put up https://github.com/PyCQA/pyflakes/pull/313 against pyflakes, which resolves this issue by actually handling forward type annotations.
